pepper-apis/src/main/java/org/broadinstitute/ddp/log/SlackAppender.java

```diff
@@ -43,45 +53,72 @@
 
     private boolean canLog = true;
 
-    /**
-     * Used for testing.  Actual deployments use {@link #SlackAppender() the no-arg constructor}
-     */
-    public SlackAppender(Slack slack, String slackHookUrl, String slackChannel) {
-        this.slack = slack;
-        this.slackHookUrl = slackHookUrl;
-        this.channel = slackChannel;
-    }
+    private List<SlackMessagePayload> messagesToSend = Collections.synchronizedList(new ArrayList<>());
 
-    /**
-     * Constructor used by deployments via logback.xml
-     */
-    public SlackAppender() {
-        slack = Slack.getInstance();
-        Config cfg = ConfigManager.getInstance().getConfig();
-        String slackHook = null;
-        String slackChannel = null;
-        if (cfg != null) {
-            if (cfg.hasPath(ConfigFile.SLACK_HOOK)) {
-                slackHook = cfg.getString(ConfigFile.SLACK_HOOK);
-            }
-            if (cfg.hasPath(ConfigFile.SLACK_CHANNEL)) {
-                slackChannel = cfg.getString(ConfigFile.SLACK_CHANNEL);
-            }
+    private int queueSize;
+
+    private int intervalInMillis;
+
+    private HttpClient httpClient;
+
+    private void init(String slackHookUrl,
+                      String slackChannel,
+                      String queueSize,
+                      String intervalInMillis) {
+        httpClient = HttpClient.newHttpClient();
+        try {
+            this.slackHookUrl = new URI(slackHookUrl);
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException("Could not parse " + slackHookUrl);
+        }
+
+        this.channel = slackChannel;
+        if (StringUtils.isNumeric(queueSize)) {
+            this.queueSize = Integer.parseInt(queueSize);
+        } else {
+            this.queueSize = 10;
+        }
+        if (StringUtils.isNumeric(intervalInMillis)) {
+            this.intervalInMillis = Integer.parseInt(intervalInMillis);
+        } else {
+            this.intervalInMillis = 60;
         }
 
-        if (StringUtils.isBlank(slackHook)) {
-            LOG.warn("No slack hook value for " + ConfigFile.SLACK_HOOK + ".  No logs will go to slack.");
+        if (StringUtils.isBlank(slackHookUrl)) {
+            LOG.warn("No logs will go to slack.");
             canLog = false;
         }
         if (StringUtils.isBlank(slackChannel)) {
-            LOG.warn("No slack channel value for " + ConfigFile.SLACK_CHANNEL + ".  No logs will go to slack.");
+            LOG.warn("No logs will go to slack.");
             canLog = false;
         }
-        this.slackHookUrl = slackHook;
-        this.channel = slackChannel;
         if (canLog) {
-            LOG.info("Slack alerts will be sent to " + slackChannel);
+            LOG.info("At most {} slack alerts will be sent to {} every {} ms", queueSize, slackChannel, intervalInMillis);
         }
+
+        ScheduledThreadPoolExecutor executorService = new ScheduledThreadPoolExecutor(1,
```

Comment:
As a small optimization, perhaps we should move initialization of `httpclient` and `executorService` inside the `if (canLog)` block above? So local development can completely turn of slack appender.

Reply:
Done!